spring-boot-project/spring-boot/src/main/java/org/springframework/boot/DigitalAmount.java

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright 2012-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot;
+
+import java.io.Serializable;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import org.springframework.lang.NonNull;
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+
+/**
+ * This class models a quantity or amount of digital information in terms of bytes. This
+ * class is immutable and thread-safe.
+ *
+ * @author Dmytro Nosan
+ * @since 2.1.0
+ * @see DigitalAmountStyle
+ * @see DigitalUnit
+ */
+public final class DigitalAmount implements Comparable<DigitalAmount>, Serializable {
+
+	/**
+	 * amount in terms of bytes.
+	 */
+	private final long amount;
```

Comment:
I'd call this `bytes`. It's not a `@ConfigurationProperties` so we don't need javadoc on that field.